template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/CmTemplateValidator.java

```diff
@@ -65,6 +68,21 @@ public void validateHostGroupScalingRequest(String accountId, Blueprint blueprin
         }
     }
 
+    private void validateNodeManagerAdjustment(HostGroup hostGroup, Integer adjustment, Collection<InstanceGroup> instanceGroups,
+            CmTemplateProcessor templateProcessor) {
+        Map<String, Set<String>> nodeManagerComponentInGroups = templateProcessor.getComponentsByHostGroup().entrySet().stream()
+                .filter(entry -> entry.getValue().contains("NODEMANAGER"))
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+        if (nodeManagerComponentInGroups.containsKey(hostGroup.getName())) {
+            instanceGroups.stream().filter(instanceGroup -> nodeManagerComponentInGroups.containsKey(instanceGroup.getGroupName()))
+                    .map(InstanceGroup::getNodeCount).reduce(Integer::sum).ifPresent(nodeManagerCountInGroups -> {
+                if (nodeManagerCountInGroups + adjustment <= 0) {
+                    throw new BadRequestException("At lease one instance required with NODEMANAGER role.");
```

Comment:
@biharitomi as I see this code works based on the blueprint and not based on the actual state in CM. Is it acceptable? If yes then I suggest to change the error message to
```
Scaling adjustment is not allowed, based on the template it would delete all nodes with required Node Manager role.
```

Reply:
Fine for me.